x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -428,10 +429,15 @@ export class TaskRunner<
       });
     }
 
+    let executedActions: AlertAction[] = [];
     if (!muteAll && this.shouldLogAndScheduleActionsForAlerts()) {
       const mutedAlertIdsSet = new Set(mutedInstanceIds);
 
-      scheduleActionsForRecoveredAlerts<InstanceState, InstanceContext, RecoveryActionGroupId>({
+      const scheduledActionsForRecoveredAlerts = await scheduleActionsForRecoveredAlerts<
```

Comment:
Since we just need the number, could we simplify this to just keep track of a running count instead of passing around the full array? Do we think we'll need the information in `AlertAction` for another reason in the future?

Reply:
Yes, we discussed this with Patrick before, i think we may need the actions for another metrics etc. in the future.
And i think returning the executed actions from the execution_handler makes more sense.